src/continuation/homotopy_optimizer.rs

```diff
@@ -332,6 +413,12 @@ where
                 self.update_continuation_parameters(&mut q_augmented_param_vec);
                 exit_status = ExitStatus::NotConvergedIterations;
             }
+
+            // Update the current inner tolerance (until the target inner tolerance in reached)
+            inner_tolerance = f64::max(
```

Comment:
I would recommend the more rustic way to do max:
```rust
inner_tolerance = target_inner_tolerance.max(self.inner_tolerance_update_factor * inner_tolerance);
```

Reply:
I agree. I implemented this.